src/accel/SharedParams.cc

```diff
@@ -219,19 +249,24 @@ SharedParams::SharedParams(SetupOptions const& options)
  */
 void SharedParams::InitializeWorker(SetupOptions const&)
 {
+    CELER_VALIDATE(!CeleritasDisabled(),
+                   << "Celeritas shared params cannot be initialized when "
+                      "Celeritas offloading is disabled via "
+                      "\"CELER_DISABLE\"");
+
     celeritas::activate_device_local();
 }
 
 //---------------------------------------------------------------------------//
 /*!
  * Clear shared data after writing out diagnostics.
  *
- * This must be executed exactly *once* across all threads and at the end of
+ * This should be executed exactly *once* across all threads and at the end of
  * the run.
  */
 void SharedParams::Finalize()
 {
-    CELER_EXPECT(*this);
+    std::lock_guard scoped_lock{updating_mutex()};
```

Comment:
Should be, yes, but I figured someone is likely to call it from all threads... is the right move to allow it safely (basically no performance penalty since we only call it once per run) or to let it be undefined or to try to catch it?

Reply:
Yeah, this way seems better.